### PR TITLE
feat(interceptors): add interceptor for `X-App` headers in launcher API calls

### PR DESCRIPTION
**Pull request by rohitkrai03**
Planner Story - https://openshift.io/openshiftio/Openshift_io/plan/detail/884

- Adds an interceptor for `X-App` headers in launcher API calls

### CODE REVIEW

**Comment by rohitkrai03** [comment]
@alien-ike I think you cannot identify a test file pattern like `something.interceptor.spec.ts`. You need to be updated. :)

**Comment by joshuawilson** [comment]
haha, we need to tell @bartoszmajsak about this

File: src/app/shared/interceptors/x-app.interceptor.spec.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    });
+    httpMock = TestBed.get(HttpTestingController);
+    httpClient = TestBed.get(HttpClient);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should add an X-App header', () => {
```
Comment (christianvogt) [inline src/app/shared/interceptors/x-app.interceptor.spec.ts]: You should have separate tests for `FABRIC8_FORGE_API_URL` and another for `ANALYTICS_RECOMMENDER_URL`.

File: src/app/shared/interceptors/x-app.interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    @Inject(FABRIC8_FORGE_API_URL) private forgeApiUrl: string,
+    @Inject(ANALYTICS_RECOMMENDER_URL) private recommenderApiUrl: string
+  ) {}
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    const url = request.url;
+
+    // Attach a X-App to all requests that don't have one
+    if (url.startsWith(this.forgeApiUrl) || url.startsWith(this.recommenderApiUrl)) {
+      if (!request.headers.has('X-App')) {
```
Comment (christianvogt) [inline src/app/shared/interceptors/x-app.interceptor.ts]: This `if` statement should be merged with the one above. Can't wait till we have prettier support.

**Pull request merged**